File: TurretOutput.cpp
#include "WPILib.h"

#define TURRET_POWER_MOD -1

class TurretOutput : public PIDOutput {
	Victor *turretMotor;
	
	public:
		float lastOutput;
		
		TurretOutput(Victor *tM) {
			turretMotor = tM;
			lastOutput = 0;
		}
		
		virtual void PIDWrite(float output) {
			output *= TURRET_POWER_MOD;
			lastOutput = output;
			turretMotor->Set(output);
		}
};

File: src/com/_604robotics/robot2012/visiontesting/TurretOutput.java
package com._604robotics.robot2012.visiontesting;

import edu.wpi.first.wpilibj.PIDOutput;
import edu.wpi.first.wpilibj.Victor;

public class TurretOutput implements PIDOutput {
    private Victor turret;
    private double lastOutput;
    
    public TurretOutput (Victor turret) {
        this.turret = turret;
    }
    
    public void pidWrite (double output) {
        output *= -1;
        this.lastOutput = output;
        turret.set(output);
    }
    
    public double getLastOutput () {
        return lastOutput;
    }
}
